mkosi/__init__.py

```diff
@@ -266,32 +266,29 @@ def run_prepare_script(state: MkosiState, build: bool) -> None:
         BUILDROOT=str(state.root),
     )
 
-    chroot: list[PathString] = chroot_cmd(
-        state.root,
-        options=[
-            "--bind", state.config.prepare_script, "/work/prepare",
-            "--bind", Path.cwd(), "/work/src",
-            "--chdir", "/work/src",
-            "--setenv", "SRCDIR", "/work/src",
-            "--setenv", "BUILDROOT", "/",
-        ],
-    )
-
-    if build:
-        with complete_step("Running prepare script in build overlay…"), mount_build_overlay(state):
-            bwrap(
-                [state.config.prepare_script, "build"],
-                network=True,
-                readonly=True,
-                options=finalize_mounts(state.config),
-                scripts={"mkosi-chroot": chroot} | package_manager_scripts(state),
-                env=env | state.config.environment,
-                stdin=sys.stdin,
+    with contextlib.ExitStack() as stack:
+        if build:
+            stack.enter_context(complete_step("Running prepare scripts in build overlay…"))
```

Comment:
I would log one step per script executed and include the script name in the step message. This makes it easier to figure out for users from which script some output is coming